### PR TITLE
fix(color-picker): avoid input/change events from firing when initializing color/value

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #2938

## Summary

This fixes an issue where the code to set the internal color from the initial value on `connectedCallback` was causing `change` and `input` event to unintentionally fire.

### CODE REVIEW

File: src/components/calcite-color-picker/calcite-color-picker.tsx [hunk 1]
```diff
@@ -294,6 +297,8 @@ export class CalciteColorPicker {
 
   private hueThumbState: "idle" | "hover" | "drag" = "idle";
 
+  private internalColorUpdateContext: "internal" | "initial" | null = null;
```
Comment (driskull) [inline src/components/calcite-color-picker/calcite-color-picker.tsx:300]: 👍

File: src/components/calcite-color-picker/calcite-color-picker.tsx [hunk 2]
```diff
@@ -696,15 +701,17 @@ export class CalciteColorPicker {
 
     const valueIsNoColor = allowEmpty && !value;
     const parsedMode = parseMode(value);
-    const valueIsCompatible = (format === "auto" && parsedMode) || format === parsedMode;
+    const valueIsCompatible =
+      valueIsNoColor || (format === "auto" && parsedMode) || format === parsedMode;
```
Comment (driskull) [inline src/components/calcite-color-picker/calcite-color-picker.tsx:705]: nitpick. Maybe const could be `valueNotAColor` or `colorlessValue`? no biggie. Just a weird name `valueIsNoColor`

**Review by driskull — APPROVED** [review]
👍 🎉

File: src/components/calcite-color-picker/calcite-color-picker.tsx (hunk 2 above)
Comment (jcfranco) [reply src/components/calcite-color-picker/calcite-color-picker.tsx:705]: Good idea. Changed it to `willSetNoColor` to better express the intent.

**Pull request merged**